File: utils/email.js
/**
 * Email sending utility.
 *
 * Sends transactional emails (form notifications, newsletter welcome mails)
 * over SMTP using the EMAIL_* settings from config.js.
 */
var nodemailer = require('nodemailer');
const ErrorResponse = require('../utils/errorResponse');
const { emailHost, emailPort, emailUsername, emailPassword } = require('../config/config');

/**
 * Sends an email via the configured SMTP account.
 *
 * @param {string|false} [email=false] - Recipient address; pass `false` to
 *   send to the SMTP account itself (used for internal team notifications).
 * @param {string} subject - Email subject line.
 * @param {string} [text=""] - Plain-text fallback body.
 * @param {string} html - HTML body.
 * @returns {Promise<boolean>} always resolves true — send failures are
 *   logged but never thrown, so an email outage can't fail the API request.
 */
exports.generateEmail = async (email = false, subject, text = "", html) => {
    try {

        const transporter = nodemailer.createTransport({
            host: emailHost,
            port: emailPort,
            auth: {
                user: emailUsername,
                pass: emailPassword
            },
            secure: false,
            tls: {
                rejectUnauthorized: false,
                minVersion: "TLSv1.2"
            }
        });

        const mailOptions = {
            from: emailUsername,
            to: email ? email : emailUsername,
            subject,
            text: text,
            html: html,
        };
        const send = await transporter.sendMail(mailOptions);


        transporter.verify(function (error, success) {
            if (error) {
                console.log(error);
            } else {
                console.log("Server is ready to take our messages");
            }
        });

        return true;
    } catch (err) {
        new ErrorResponse("Error in generating email", 400)
        console.log("err in generate email: ", err);
        return true;
    }
}

// this.generateEmail('newgen@mailinator.com', 'MARCH', 'details')

// this.generateEmail('jahakiy676@wikfee.com', 'this is the testing email', `<h1>hello</h1>`)
// this.generateEmail('shayanmustafa11@gmail.com', 'this is the testing email', `<h1>hello</h1>`)

File: utils/email.py
"""
Email sending utility.

Sends transactional emails (form notifications, newsletter welcome mails)
over SMTP using the EMAIL_* settings from config.py. Uses the standard
library only (smtplib/email), mirroring the nodemailer setup in the
original app: STARTTLS with a minimum TLS version of 1.2.
"""
import smtplib
import ssl
from email.mime.multipart import MIMEMultipart
from email.mime.text import MIMEText

from config.config import config


def generate_email(email=False, subject='', text='', html='') -> bool:
    """Sends an email via the configured SMTP account.

    :param email: recipient address; pass False/None to send to the SMTP
        account itself (used for internal team notifications).
    :param subject: email subject line.
    :param text: plain-text fallback body.
    :param html: HTML body.
    :return: always True — send failures are logged but never raised, so an
        email outage can't fail the API request.
    """
    try:
        message = MIMEMultipart('alternative')
        message['From'] = config.EMAIL_USERNAME
        message['To'] = email if email else config.EMAIL_USERNAME
        message['Subject'] = subject

        if text:
            message.attach(MIMEText(text, 'plain'))
        if html:
            message.attach(MIMEText(html, 'html'))

        tls_context = ssl.create_default_context()
        tls_context.minimum_version = ssl.TLSVersion.TLSv1_2
        # mirrors `tls: { rejectUnauthorized: false }` in the original —
        # accepts the SMTP host's certificate even if it can't be verified
        tls_context.check_hostname = False
        tls_context.verify_mode = ssl.CERT_NONE

        with smtplib.SMTP(config.EMAIL_HOST, config.EMAIL_PORT) as server:
            server.starttls(context=tls_context)
            server.login(config.EMAIL_USERNAME, config.EMAIL_PASSWORD)
            server.sendmail(config.EMAIL_USERNAME, message['To'], message.as_string())

        print('Email sent successfully')
        return True
    except Exception as err:
        print('err in generate email: ', err)
        return True
